_posts/2017-08-16-json-server.md

```diff
@@ -0,0 +1,655 @@
+    "article": {
+      "id": 2,
+      "title": "Build an API using API Platform",
+      "author": "rpierlot"
+    }
+  },
+  ...
+]
+```
+
+Until now, we've seen only the routes part of `json-server`, there are a lot more things we can do, let's see what's next.
+
+<a name="random-data"></a>
+
+### Randomly generated data
+
+The [basic example in Typicode's docs](https://github.com/typicode/json-server#generate-random-data) presents a simple script that generates the `users` endpoint. Here we are going to write endpoints that serve randomly generated data using a data faker module. Personally, I use [faker.js](https://github.com/Marak/faker.js), but there are others that you can explore like [Chance](https://github.com/chancejs/chancejs) and [Casual](https://github.com/boo1ean/casual).
+
+The random aspect of the generation occurs only once, and that's when we run the script. This means that `json-server` won't give us a different response for each request. Eventually, we have to install our fake data tool, then write the generation script:
+
+```bash
+$ yarn add faker
+$ touch generate.js
+```
+
+Keep in mind that the script must export a function that exclusively returns an object with keys (endpoints) inside.
+
+```js
+// generate.js
+const faker = require('faker');
+
+module.exports = () => ({
+  messages: [...Array(3)].map((value, index) => ({
+    id: index + 1,
+    name: faker.hacker.noun(),
+    status: faker.hacker.adjective(),
+    description: faker.hacker.phrase(),
+  })),
+});
+```
+
+Then run json-server by giving it the generation script:
+
+```bash
+$ json-server generate.js
+
+  \{^_^}/ hi!
+
+  Loading generate.js
+  Done
+
+  Resources
+  http://localhost:3000/messages
+
+  Home
+  http://localhost:3000
+
+  Type s + enter at any time to create a snapshot of the database
+```
+
+And the results would be something like:
+
+```bash
+$ curl http://localhost:3000/messages
+[
+  {
+    "id": 1,
+    "name": "driver",
+    "status": "cross-platform",
+    "description": "If we connect the system, we can get to the ADP panel through the redundant PCI protocol!"
+  },
+  {
+    "id": 2,
+    "name": "monitor",
+    "status": "1080p",
+    "description": "Try to synthesize the CSS driver, maybe it will navigate the bluetooth matrix!"
+  },
+  {
+    "id": 3,
+    "name": "hard drive",
+    "status": "virtual",
+    "description": "Use the redundant SMS program, then you can compress the bluetooth port!"
+  }
+]
+```
+
+And we can still perform requests on it as we've seen in the [routes](#routes) section.
+
+<a name="custom-routes"></a>
+
+### Custom routes
+
+Imagine we are supposed to perform requests on several different endpoints on our future API, and these endpoints don't have the same URIs:
+
+```url
+/api/dashboard
+/api/groups/ducks/stats
+/auth/users
+/rpierlot/articles
+```
+
+`json-server` allows us to specify route rewrites. We can address this problem using a map that resolves the actual routes in our json schema:
```

Comment:
+`json-server` allows us to specify route rewrites. We can address this problem by using a map that resolves the actual routes in our json schema: